devito/mpi/routines.py

```diff
@@ -1058,6 +1058,7 @@ def __init__(self, name, key, arguments, owned):
         name = "%s%d" % (name, key)
```

Comment:
Actually...  isn't this going to make it different when it's unpickled?  Since the name property is no longer the name passed to init?

Reply:
oohhh right nice catch! I even wrote an MFE for MPIMsg but not for MPIRegion. 

should be a trivial fix, will do on Monday